### PR TITLE
Add use-cases and components sections

### PR DESCRIPTION
**Pull request by lukekim**
## 🗣 Description

Add new Use Cases and Components sections.

Move Data Accelerators, Data Connectors, and Secret Stores into Components.

### CODE REVIEW

File: spiceaidocs/docs/components/data-accelerators/duckdb.md [hunk 1]
```diff
@@ -40,9 +40,5 @@ datasets:
...
     - `SELECT [1, 2, 3];`
     - `SELECT ['1992-09-20 11:30:00.123456789', 'epoch'::TIMESTAMP]`
-  - Unsupported:
-    - `SELECT [['duck', 'goose', 'heron'], ['frog', 'toad']]`
-    - `SELECT {'x': 1, 'y': 2, 'z': 3}`
-    - `SELECT MAP(['key1', 'key2', 'key3'], [10, 20, 30])`
-    - `SELECT ['duck', 'goose', 'heron'];`
-:::
\ No newline at end of file
+  - Unsupported: - `SELECT [['duck', 'goose', 'heron'], ['frog', 'toad']]` - `SELECT {'x': 1, 'y': 2, 'z': 3}` - `SELECT MAP(['key1', 'key2', 'key3'], [10, 20, 30])` - `SELECT ['duck', 'goose', 'heron'];`
```
Comment (y-f-u) [inline spiceaidocs/docs/components/data-accelerators/duckdb.md]: lines?

File: spiceaidocs/docs/components/data-connectors/s3.md [hunk 2]
```diff
@@ -165,5 +168,6 @@ Create a dataset named `taxi_trips` from a public S3 folder.
 ```
 
 :::warning[Limitations]
+
 - Only S3 authentication using `aws_access_key_id` and `aws_secret_access_key` is currently supported. Please [submit an issue](https://github.com/spiceai/spiceai/issues/new?template=feature_request.md) if you would like to see support for other authentication methods.
-:::
+  :::
```
Comment (y-f-u) [inline spiceaidocs/docs/components/data-connectors/s3.md]: the indentation is off

File: spiceaidocs/docs/features/cdc/index.md [hunk 3]
```diff
@@ -23,13 +23,13 @@ When configuring datasets to be accelerated with CDC, ensure that the data conne
 
 Startup time for CDC-accelerated datasets may be longer than non-CDC-accelerated datasets due to the initial synchronization of the dataset.
 
-It's recommended to use CDC-accelerated datasets with persistent data accelerator configurations (i.e. `file` mode for [`DuckDB`](../../data-accelerators/duckdb.md)/[`Sqlite`](../../data-accelerators/sqlite.md) or [`PostgreSQL`](../../data-accelerators/postgres/index.md)). This ensures that when Spice restarts, it can resume from the last known state of the dataset instead of re-fetching the entire dataset.
+It's recommended to use CDC-accelerated datasets with persistent data accelerator configurations (i.e. `file` mode for [`DuckDB`](/components/data-accelerators/duckdb.md)/[`Sqlite`](/components/data-accelerators/duckdb.md) or [`PostgreSQL`](/components/data-accelerators/postgres/index.md)). This ensures that when Spice restarts, it can resume from the last known state of the dataset instead of re-fetching the entire dataset.
```
Comment (y-f-u) [inline spiceaidocs/docs/features/cdc/index.md]: duplicated duckdb.md

File: spiceaidocs/docs/features/local-acceleration/constraints.md [hunk 4]
```diff
@@ -42,7 +42,7 @@ Examples
 
 ## Limitations
 
-- **Not supported for in-memory Arrow:** The default in-memory Arrow acceleration engine does not support constraints. Use [DuckDB](../../data-accelerators/duckdb.md), [SQLite](../../data-accelerators/sqlite.md), or [PostgreSQL](../../data-accelerators/postgres/index.md) as the acceleration engine to enable constraint checking.
+- **Not supported for in-memory Arrow:** The default in-memory Arrow acceleration engine does not support constraints. Use [DuckDB](/components/data-accelerators/duckdb.md), [SQLite](/components/data-accelerators/duckdb.md), or [PostgreSQL](/components/data-accelerators/postgres/index.md) as the acceleration engine to enable constraint checking.
```
Comment (y-f-u) [inline spiceaidocs/docs/features/local-acceleration/constraints.md:45]: duplicated duckdb.

File: spiceaidocs/docs/features/local-acceleration/indexes.md [hunk 5]
```diff
@@ -34,11 +34,12 @@ Examples
...
   - Similar to specifying `CREATE INDEX my_index ON my_table (my_column)`.
 - `unique`: Creates a unique index on the specified column(s). See [Constraints](./constraints.md) for more information on working with unique constraints on locally accelerated tables.
   - Similar to specifying `CREATE UNIQUE INDEX my_index ON my_table (my_column)`.
 
 :::warning[Limitations]
-- **Not supported for in-memory Arrow:** The default in-memory Arrow acceleration engine does not support indexes. Use [DuckDB](../../data-accelerators/duckdb.md), [SQLite](../../data-accelerators/sqlite.md), or [PostgreSQL](../../data-accelerators/postgres/index.md) as the acceleration engine to enable indexing.
-:::
\ No newline at end of file
+
+- **Not supported for in-memory Arrow:** The default in-memory Arrow acceleration engine does not support indexes. Use [DuckDB](/components/data-accelerators/duckdb.md), [SQLite](/components/data-accelerators/duckdb.md), or [PostgreSQL](/components/data-accelerators/postgres/index.md) as the acceleration engine to enable indexing.
```
Comment (y-f-u) [inline spiceaidocs/docs/features/local-acceleration/indexes.md:44]: duplicatd duckdb.md

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**